src/super_context.ml

```diff
@@ -859,23 +870,53 @@ module PP = struct
         | driver :: rest ->
           (Some driver, List.sort rest ~cmp:String.compare @ [driver])
       in
-      build_ppx_driver sctx names ~dir:ppx_dir ~dep_kind:Required ~target:exe ~driver
+      build_ppx_driver sctx names ~dir:scope_dir ~dep_kind:Required ~target:exe ~driver
     | _ -> ()
 
-  let get_ppx_driver sctx pps =
-    let names =
+  let get_ppx_driver sctx ~(scope:Jbuild.Scope.t) pps =
+    let (driver, names) =
       match List.rev_map pps ~f:Pp.to_string with
-      | [] -> []
-      | driver :: rest ->
-        List.sort rest ~cmp:String.compare @ [driver]
+      | [] -> (None, [])
+      | driver :: rest -> (Some driver, List.sort ~cmp:String.compare rest)
     in
+    let sctx = host_sctx sctx in
+    let from = Libs.find_scope_dir_by_name_exn sctx (Scope.name scope) in
+    let public_name name =
+      match Libs.find sctx ~from name with
+      | None -> Some name (* XXX unknown but assume it's public *)
+      | Some lib -> Lib.public_name lib in
+    let (driver_public, driver_name) =
+      match driver with
+      | None -> (true, None)
+      | Some driver ->
+        begin match public_name driver with
+        | None -> (false, Some driver)
+        | Some driver -> (true, Some driver)
+        end in
+    let (public, private_) =
+      List.fold_left ~f:(fun (pubs, privs) name ->
```

Comment:
I think the key end up a bit weird: inversely sorted list of public names followed by sorted list of private names. What about keeping all names in the same list and folding a `has_private_libs` boolean?